Index: vendor/ffmpeg-win32-dev/include/libavutil/mem.h
===================================================================
--- vendor/ffmpeg-win32-dev/include/libavutil/mem.h	(revision 94192)
+++ vendor/ffmpeg-win32-dev/include/libavutil/mem.h	(working copy)
@@ -69,7 +69,7 @@
 #if AV_GCC_VERSION_AT_LEAST(4,3)
     #define av_alloc_size(...) __attribute__((alloc_size(__VA_ARGS__)))
 #else
-    #define av_alloc_size(...)
+    #define av_alloc_size(n)
 #endif
 
 /**
